Add configuration example for connecting to sql server named instances (#16247)
DataDog/documentation: layouts/shortcodes/dbm-sqlserver-agent-config-examples.md

    password: '<PASSWORD>'
    reported_hostname: products-replica-1
```

> ### Using port autodiscovery

Reviewer: ```suggestion
### Discovering ports automatically
```

changing this to disambiguate from container autodiscovery

Author: @cswatt this is verbiage specific to SQL Server, why change it?

> Some services like SQL Server Browser Service or Named Instances allow the user to not worry about hardcoded port numbers in the connection string. In order to use the agent with one of these services, the `port` field should be set to `0`.

Reviewer: ```suggestion
SQL Server Browser Service, Named Instances, and other services can automatically detect port numbers. You can use this instead of hardcoding port numbers in connection strings. To use the Agent with one of these services, set the `port` field to `0`.
```
